### ISSUE TITLE
missing_safety_doc should catch `unsafe impl`

### ISSUE THREAD

**Issue by sffc** [rust-lang/rust-clippy#7732]
I tried this code:

```rust
struct ExampleStruct {}
unsafe trait ExampleUnsafeTrait {}

unsafe impl ExampleUnsafeTrait for ExampleStruct {}
```

I expected to see this happen: The `missing_safety_doc` should complain about the `unsafe impl` not having safety docs.

Instead, this happened: This code passes the clippy check.

### Meta

**Rust version (`rustc -Vv`):**

```
rustc 1.56.0-beta.3 (deef86610 2021-09-17)
binary: rustc
commit-hash: deef866109022248b2506587b377c6b1c881d963
commit-date: 2021-09-17
host: x86_64-unknown-linux-gnu
release: 1.56.0-beta.3
LLVM version: 13.0.0
```

CC @Manishearth

**Comment by Manishearth** [comment]
`missing_safety_doc` is about documenting safety invariants for library types, _not_ for documenting `unsafe` code in comments. As such it's irrelevant when it comes to `unsafe impl`s (or `unsafe` blocks). However, it should absolutely warn on `unsafe trait` so we can fix that.

**Comment by Manishearth** [comment]
https://github.com/rust-lang/rust-clippy/pull/7734
